### ISSUE TITLE
Crash when parsing non-standard date values.

### ISSUE THREAD

**Issue by notnotmelon** [fgardt/factorio-mod-template#2]
I am adding this template into existing projects with non-standard date formats in the changelog.

Example:
```lua
---------------------------------------------------------------------------------------------------
Version: 1.2.0
Date: 9.10.2020
  Features:
    - Fixed that units would sometimes go inactive
    - Changed some descriptions
    - Tempature is now stored
    - You can no longer fill the units with inserters or loaders
    - Increased power usage
    - Space ex compaiblity
    - Fluid texture is now animated!
---------------------------------------------------------------------------------------------------
Version: 1.1.0
Date: 9.9.2020
  Features:
    - Removed dependancy on main mod
    - Fixed a crash
    - Fixed an issue with the fluid display
    - You can now use memory elements with fluid units
    - Adjusted pressure
    - Reduced power usage
    - Added a setting to adjust power usage
    - Fixed some issues with shadows
---------------------------------------------------------------------------------------------------
Version: 1.0.0
Date: 9.2.2020
  Features:
    - Initial release
```

Semantic release crashes with
```
 TypeError: Method Date.prototype.toString called on incompatible receiver [object Date]
 ```
 
 The full logs are visible here: https://github.com/notnotmelon/fluid-memory-storage/actions/runs/10228842045/job/28301807293
 
## Suggested fix

 Perhaps instead catch the exception and create a more useful message explaining the intended date format.

**Comment by fgardt** [comment]
the error is unrelated to the used date format in existing changelog entries (existing changelog is not even looked at, only appended to).

I am already in the progress of fixing this, thanks for notifying me tho :)

**Comment by fgardt** [comment]
as a temporary workaround change your `.github/workflows/release.yml` to this

```yml
name: Release
permissions: write-all
on:
  push:
    branches:
      - main
jobs:
  release:
    name: release
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3
      - name: Set up Node.js
        uses: actions/setup-node@v3
        with:
          node-version: lts/*
      - name: Install release dependencies
        run: |
          npm install semantic-release@^20.0.2 \
          @semantic-release/git@^10.0.1 semantic-release-factorio@^1.5.0 \
          conventional-changelog-conventionalcommits-factorio@^1.0.3
      - name: Run semantic-release
        run: npx semantic-release
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
FACTORIO_TOKEN: ${{ secrets.FACTORIO_TOKEN }}
```

**Comment by fgardt** [comment]
This should now be solved with 0d9d2b4b7ca14d1370d35d24f71c5912e00f0d06.

To fix your repository copy the contents of [`.releaserc`](https://github.com/fgardt/factorio-mod-template/tree/0d9d2b4b7ca14d1370d35d24f71c5912e00f0d06/.releaserc) and [`release.yml`](https://github.com/fgardt/factorio-mod-template/tree/0d9d2b4b7ca14d1370d35d24f71c5912e00f0d06/.github/workflows/release.yml)
